Polyhedron/demo/Polyhedron/Plugins/Mesh_3/Mesh_3_plugin.cpp

```diff
@@ -182,66 +176,76 @@ void Mesh_3_plugin::mesh_3_volume()
 
 void Mesh_3_plugin::mesh_3(const bool surface_only, const bool use_defaults)
 {
-  Scene_surface_mesh_item* sm_item = NULL;
-  Scene_surface_mesh_item* bounding_sm_item = NULL;
-  Scene_implicit_function_item* function_item = NULL;
-  Scene_image_item* image_item = NULL;
-  Scene_polylines_item* polylines_item = NULL;
-
-  Q_FOREACH(int ind, scene->selectionIndices()) {
-    if(sm_item == NULL)
-    {
-      sm_item = qobject_cast<Scene_surface_mesh_item*>(scene->item(ind));
-      if (sm_item != NULL
-          && scene->selectionIndices().size() == 2
-          && bounding_sm_item == NULL)
-      {
-        bounding_sm_item = qobject_cast<Scene_surface_mesh_item*>(
-            scene->item(scene->selectionIndices().back()));
-        if (bounding_sm_item != NULL)
-        {
-          if (is_closed(*sm_item->polyhedron())
-            && !is_closed(*bounding_sm_item->polyhedron()))
-          {
-            //todo : check sm_item is inside bounding_sm_item
-            std::swap(sm_item, bounding_sm_item);
-            //now bounding_sm_item is the bounding one
-          }
-        }
+  QList<Scene_surface_mesh_item*> sm_items;
+  Scene_surface_mesh_item* bounding_sm_item = nullptr;
+  Scene_implicit_function_item* function_item = nullptr;
+  Scene_image_item* image_item = nullptr;
+  Scene_polylines_item* polylines_item = nullptr;
+
+  for(int ind: scene->selectionIndices()) {
+    Scene_surface_mesh_item* sm_item =
+      qobject_cast<Scene_surface_mesh_item*>(scene->item(ind));
+    if(sm_item) {
+      sm_items.push_back(sm_item);
+      if(is_closed(*sm_item->polyhedron())) {
+        bounding_sm_item = sm_item;
       }
     }
 #ifdef CGAL_MESH_3_DEMO_ACTIVATE_IMPLICIT_FUNCTIONS
-    if(function_item == NULL){
-      function_item = qobject_cast<Scene_implicit_function_item*>(scene->item(ind));
-    }
+    else if(function_item == nullptr &&
+            nullptr !=
+            (function_item = qobject_cast<Scene_implicit_function_item*>(scene->item(ind))))
+    {}
 #endif
 #ifdef CGAL_MESH_3_DEMO_ACTIVATE_SEGMENTED_IMAGES
-    if(image_item == NULL){
-      image_item = qobject_cast<Scene_image_item*>(scene->item(ind));
-    }
+    else if(image_item == nullptr &&
+            nullptr != (image_item = qobject_cast<Scene_image_item*>(scene->item(ind))))
+    {}
 #endif
-    if(polylines_item == NULL){
-      polylines_item = qobject_cast<Scene_polylines_item*>(scene->item(ind));
+    else if(polylines_item == nullptr &&
+            nullptr != (polylines_item = qobject_cast<Scene_polylines_item*>(scene->item(ind))))
+    {}
+    else {
+      QMessageBox::warning(mw, tr("Mesh_3 plugin"),
+                           tr("Wrong selection of items"));
+      return;
     }
   }
-  Scene_item* item = NULL;
+  Scene_item* item = nullptr;
+  bool more_than_one_item = false;
```

Comment:
This is left unused and leads to warnings in the testsuite

Reply:
That is right. I forgot about it! I am glad the compiler remind me on that part of the changes I wanted to make... I'll fix it now...